### PR TITLE
locations time frame restriction

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3242656&stafilter=NotComplete&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Models/Foundation/Summit/Locations/SummitVenueRoom.php [hunk 1]
```diff
@@ -238,4 +238,21 @@ public function setFloor(SummitVenueFloor $floor)
         $this->floor = $floor;
     }
 
+    /**
+     * @return int|null
+     */
+    public function getOpeningHour(): ?int
+    {
+        return $this->opening_hour == null ?
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Locations/SummitVenueRoom.php:246]: you could use ?? operator here

File: database/migrations/model/Version20230425143939.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema): void
+    {
+        $builder = new Builder($schema);
+        if ($schema->hasTable("SummitAbstractLocation")) {
+            $builder->table('SummitAbstractLocation', function (Table $table) {
+                $table->unsignedSmallInteger('OpeningHour')->setNotnull(false);
+                $table->unsignedSmallInteger('ClosingHour')->setNotnull(false);
```
Comment (smarcet) [inline database/migrations/model/Version20230425143939.php:35]: add default value NULL

File: app/Http/Controllers/Apis/Protected/Summit/Factories/SummitAbstractLocationValidationRulesFactory.php [hunk 3]
```diff
@@ -27,17 +27,21 @@ public static function build(array $data, $update = false){
...
             return [
-                'name'        => 'sometimes|string|max:255',
-                'short_name'        => 'sometimes|string|max:255',
-                'description' => 'sometimes|string',
-                'order'       => 'sometimes|integer|min:1'
+                'name'         => 'sometimes|string|max:255',
+                'short_name'   => 'sometimes|string|max:255',
+                'description'  => 'sometimes|string',
+                'opening_hour' => 'sometimes|integer|min:0|max:2359',
+                'closing_hour' => 'sometimes|integer|gt:opening_hour|min:0|max:2359',
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/Factories/SummitAbstractLocationValidationRulesFactory.php:34]: you need https://laravel.com/docs/10.x/validation#rule-required-with here 
like required_with:opening_hour

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Models/Foundation/Summit/Locations/SummitVenueRoom.php (hunk 1 above)
Comment (romanetar) [reply app/Models/Foundation/Summit/Locations/SummitVenueRoom.php]: done

File: app/Models/Foundation/Summit/Locations/SummitVenueRoom.php [hunk 4]
```diff
@@ -238,4 +238,21 @@ public function setFloor(SummitVenueFloor $floor)
...
+            $this->venue->getOpeningHour() : $this->opening_hour;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getClosingHour(): ?int
+    {
+        return $this->closing_hour == null ?
```
[the remark this answers is not in the document]
Comment (romanetar) [reply app/Models/Foundation/Summit/Locations/SummitVenueRoom.php]: done

File: database/migrations/model/Version20230425143939.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema): void
+    {
+        $builder = new Builder($schema);
+        if ($schema->hasTable("SummitAbstractLocation")) {
+            $builder->table('SummitAbstractLocation', function (Table $table) {
+                $table->unsignedSmallInteger('OpeningHour')->setNotnull(false);
```
[the remark this answers is not in the document]
Comment (romanetar) [reply database/migrations/model/Version20230425143939.php]: done

File: database/migrations/model/Version20230425143939.php (hunk 2 above)
Comment (romanetar) [reply database/migrations/model/Version20230425143939.php]: done

File: app/Http/Controllers/Apis/Protected/Summit/Factories/SummitAbstractLocationValidationRulesFactory.php (hunk 3 above)
Comment (romanetar) [reply app/Http/Controllers/Apis/Protected/Summit/Factories/SummitAbstractLocationValidationRulesFactory.php]: done

**Comment by romanetar** [comment]
@smarcet all review comments were addressed

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**